### DISCUSSION TITLE
Prometheus has high memory usage problem

### DISCUSSION THREAD

**Discussion by catkkt** [prometheus/prometheus#14169] [Q&A]
### What did you do?
We deployed prometheus(2.47.2 version) to monitor a series of  metrics,but running for a while we find it always has a high memory usage(almost 90% and our machine is 4C8G).After reading offical documents,we realized prometheus drops collected metrics datas from memory to disk after a default of 2 hours.So memory is occupied by a large number of datas in the 2 hours.    

### What did you expect to see?
Maybe there is a configuration parameter that we don't know to change the default time, could you have any suggestion?

Looking forward to your reply,thanks.

**Comment by machine424** [comment]
This seems more like a support query. I'll convert this into a discussion. For such questions, you might find it helpful to use [Slack](https://prometheus.io/community/), as the community is more active there.
Additionally, I recommend upgrading to a newer version, as numerous memory-related optimizations have been implemented since `2.47.2`.
